client/vscode/src/backend/requestGraphQl.ts

```diff
@@ -29,21 +29,21 @@ export const requestGraphQLFromVSCode = async <R, V = object>(
     const apiURL = `${GRAPHQL_URI}${nameMatch ? '?' + nameMatch[1] : ''}`
     // load custom headers from user setting if any
     const customHeaders = endpointRequestHeadersSetting()
-    // return empty array if no custom header is provided in configuration
-    const headers: HeadersInit = Object.entries(customHeaders)
+    // create a headers container based on the custom headers configuration if there are any
+    const headers = new Headers(customHeaders as HeadersInit)
     const sourcegraphURL = overrideSourcegraphURL || endpointSetting()
     const accessToken = accessTokenSetting()
 
     // Add Access Token to request header
     // Used to validate access token.
     if (overrideAccessToken) {
```

Comment:
```suggestion
    if (accessToken) {
        headers.set('Authorization', `token ${accessToken}`)
    }
```
Then remove everything in the else if statement?